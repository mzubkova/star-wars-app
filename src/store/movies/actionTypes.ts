export enum ActionTypes {
  GET_PLANETS = "@@movies/GET_PLANETS",
  GET_PLANETS_SUCCESS = "@@movies/GET_PLANETS_SUCCESS",
  GET_PLANETS_ERROR = "@@movies/GET_PLANETS_ERROR",
  GET_PEOPLE = "@@movies/GET_PEOPLE",
  GET_PEOPLE_SUCCESS = "@@movies/GET_PEOPLE_SUCCESS",
  GET_PEOPLE_ERROR = "@@movies/GET_PEOPLE_ERROR",
  GET_FILMS = "@@movies/GET_FILMS",
  GET_FILMS_SUCCESS = "@@movies/GET_FILMS_SUCCESS",
  GET_FILMS_ERROR = "@@movies/GET_FILMS_ERROR",
  SAVE_PATHNAME = "@@movies/SAVE_PATHNAME",
}
