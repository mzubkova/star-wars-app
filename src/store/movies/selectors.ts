import { createSelector } from "reselect";
import { ApplicationState } from "../types";
import * as Types from "./types";

export const getMoviesState = (state: ApplicationState): Types.IMoviesStore =>
  state.movies;

export const getPlanetsState = createSelector(
  getMoviesState,
  (movies: Types.IMoviesStore): Array<Types.TPlanet> => movies.planets
);

export const getPeopleState = createSelector(
  getMoviesState,
  (movies: Types.IMoviesStore): Array<Types.TPerson> => movies.people
);

export const getFilmsState = createSelector(
  getMoviesState,
  (movies: Types.IMoviesStore): Array<Types.TFilm> => movies.films
);

export const getMoviesIsLoading = createSelector(
  getMoviesState,
  (movies: Types.IMoviesStore): boolean => movies.isLoading
);

export const getPathname = createSelector(
  getMoviesState,
  (movies: Types.IMoviesStore): Array<string> => movies.pathname
);
