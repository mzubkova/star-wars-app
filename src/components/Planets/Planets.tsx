import React from "react";
import { TPlanet } from "../../store/movies/types";
import { StBox, StBoxItem } from "../styled/common";
import SkeletonItem from "../Skeletons";

export type TProps = {
  planets: Array<TPlanet>;
  isLoading: boolean;
};

const Planets = ({ planets, isLoading }: TProps) => (
  <StBox>
    {isLoading ? (
      <SkeletonItem />
    ) : (
      planets.map((planet) => (
        <StBoxItem key={planet.name}>
          <div>Name: {planet.name}</div>
          <div>Climate: {planet.climate}</div>
          <div>Orbital Period: {planet.orbital_period}</div>
          <div>Population: {planet.population}</div>
        </StBoxItem>
      ))
    )}
  </StBox>
);

export default Planets;
