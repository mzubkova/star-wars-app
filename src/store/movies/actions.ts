import { action } from "typesafe-actions";
import { ActionTypes as AT } from "./actionTypes";
import { TPlanet, TPerson, TFilm } from "./types";

export const getPlanets = () => action(AT.GET_PLANETS);
export const getPlanetsSuccess = (payload: Array<TPlanet>) =>
  action(AT.GET_PLANETS_SUCCESS, payload);
export const getPlanetsError = (payload: string) =>
  action(AT.GET_PLANETS_ERROR, payload);
export const getPeople = () => action(AT.GET_PEOPLE);
export const getPeopleSuccess = (payload: Array<TPerson>) =>
  action(AT.GET_PEOPLE_SUCCESS, payload);
export const getPeopleError = (payload: string) =>
  action(AT.GET_PEOPLE_ERROR, payload);
export const getFilms = () => action(AT.GET_FILMS);
export const getFilmsSuccess = (payload: Array<TFilm>) =>
  action(AT.GET_FILMS_SUCCESS, payload);
export const getFilmsError = (payload: string) =>
  action(AT.GET_FILMS_ERROR, payload);
export const savePathname = (payload: string) =>
  action(AT.SAVE_PATHNAME, payload);
