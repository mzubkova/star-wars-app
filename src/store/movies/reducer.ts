import { Reducer } from "redux";
import { ActionTypes as AT } from "./actionTypes";
import * as Types from "./types";
import { HYDRATE } from "next-redux-wrapper";

export const initialState: Types.IMoviesStore = {
  planets: [],
  people: [],
  films: [],
  isLoading: false,
  error: "",
  pathname: [],
};

export const moviesReducer: Reducer<
  Types.IMoviesStore,
  Types.TMoviesActions
> = (state = initialState, action) => {
  switch (action.type) {
    case HYDRATE:
      return { ...state };
    case AT.GET_PLANETS:
      return {
        ...state,
        isLoading: true,
      };
    case AT.GET_PLANETS_SUCCESS:
      return {
        ...state,
        planets: action.payload,
        isLoading: false,
      };
    case AT.GET_PLANETS_ERROR:
      return {
        ...state,
        isLoading: false,
        error: action.payload,
      };
    case AT.GET_PEOPLE:
      return {
        ...state,
        isLoading: true,
      };
    case AT.GET_PEOPLE_SUCCESS:
      return {
        ...state,
        people: action.payload,
        isLoading: false,
      };
    case AT.GET_PEOPLE_ERROR:
      return {
        ...state,
        isLoading: false,
        error: action.payload,
      };
    case AT.GET_FILMS:
      return {
        ...state,
        isLoading: true,
      };
    case AT.GET_FILMS_SUCCESS:
      return {
        ...state,
        films: action.payload,
        isLoading: false,
      };
    case AT.GET_FILMS_ERROR:
      return {
        ...state,
        isLoading: false,
        error: action.payload,
      };
    case AT.SAVE_PATHNAME:
      return {
        ...state,
        pathname: [...state.pathname, action.payload],
      };
    default:
      return state;
  }
};
