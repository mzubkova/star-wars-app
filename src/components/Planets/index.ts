import { connect } from "react-redux";
import Planets from "./Planets";
import { ApplicationState } from "../../store/types";
import { getMoviesIsLoading } from "../../store/movies/selectors";

const mapStateToProps = (state: ApplicationState) => ({
  isLoading: getMoviesIsLoading(state),
});

export default connect(mapStateToProps)(Planets);
