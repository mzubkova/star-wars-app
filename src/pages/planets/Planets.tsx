import React from "react";
import Planets from "../../components/Planets";
import { wrapper } from "../../store/store";
import { getPlanets } from "../../store/movies/actions";
import { TPlanet } from "../../store/movies/types";

export type TProps = {
  planets: Array<TPlanet>;
};

const PlanetsPage = ({ planets }: TProps) => {
  return <Planets planets={planets} />;
};

PlanetsPage.getInitialProps = wrapper.getInitialPageProps((store) => () => {
  store.dispatch(getPlanets());
});

export default PlanetsPage;
