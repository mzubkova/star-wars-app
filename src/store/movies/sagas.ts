import { put, takeEvery, call } from "redux-saga/effects";
import { ActionTypes as AT } from "./actionTypes";
import { SagaIterator } from "@redux-saga/core";
import * as actions from "./actions";
import * as requests from "../../api/index";

export function* getPlanetsWorker(): SagaIterator {
  try {
    const data = yield call(requests.getPlanets);
    yield put(actions.getPlanetsSuccess(data));
  } catch (e) {
    yield put(actions.getPlanetsError("Error"));
  }
}

export function* getPeopleWorker(): SagaIterator {
  try {
    const data = yield call(requests.getPeople);
    yield put(actions.getPeopleSuccess(data));
  } catch (e) {
    yield put(actions.getPeopleError("Error"));
  }
}

export function* getFilmsWorker(): SagaIterator {
  try {
    const data = yield call(requests.getFilms);
    yield put(actions.getFilmsSuccess(data));
  } catch (e) {
    yield put(actions.getFilmsError("Error"));
  }
}

export default function* actionWatcher() {
  yield takeEvery(AT.GET_PLANETS, getPlanetsWorker);
  yield takeEvery(AT.GET_PEOPLE, getPeopleWorker);
  yield takeEvery(AT.GET_FILMS, getFilmsWorker);
}
