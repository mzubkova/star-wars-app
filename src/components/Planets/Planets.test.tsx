import * as React from "react";
import Planets, { TProps } from "./Planets";
import PlanetsWr from "../Planets/index";
import { shallowRender, mockStore } from "../../../__tests__/helpers";
import { planetData } from "../../../__mocks__/data";
import { initialState } from "../../store/movies/reducer";

describe("Planets", () => {
  let initialStore = {};
  let component: any;
  let props: TProps;
  beforeEach(() => {
    props = {
      planets: planetData,
      isLoading: false,
    };
    initialStore = mockStore({ movies: initialState });
    component = shallowRender(<Planets {...props} />, initialStore);
  });
  it("should render without crashing", () => {
    expect(component).toMatchSnapshot();
  });
  it("should render Skeleton", () => {
    component = shallowRender(<Planets {...props} isLoading />, initialStore);
    expect(component).toMatchSnapshot();
  });
  it("should render wrapper", () => {
    component = shallowRender(<PlanetsWr {...props} />, initialStore);
    expect(component).toMatchSnapshot();
  });
});
