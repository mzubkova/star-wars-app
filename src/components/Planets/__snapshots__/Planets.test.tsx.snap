// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets should render Skeleton 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="common__StBox-sc-16ufsj2-0 lcfJTu"
      >
        <div
          class="common__StBoxItem-sc-16ufsj2-1 elAStK"
        >
          <div>
            Name: 
            Tatooine
          </div>
          <div>
            Climate: 
            Arid
          </div>
          <div>
            Orbital Period: 
            304
          </div>
          <div>
            Population: 
            120000
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="common__StBox-sc-16ufsj2-0 lcfJTu"
      >
        <svg
          aria-labelledby="v0huufs-aria"
          height="500"
          role="img"
          viewBox="0 0 500 500"
          width="500"
        >
          <title
            id="v0huufs-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#v0huufs-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#v0huufs-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="v0huufs-diff"
            >
              <rect
                height="100"
                rx="3"
                ry="3"
                width="500"
                x="0"
                y="0"
              />
              <rect
                height="100"
                rx="3"
                ry="3"
                width="500"
                x="0"
                y="115"
              />
              <rect
                height="100"
                rx="3"
                ry="3"
                width="500"
                x="0"
                y="230"
              />
              <rect
                height="100"
                rx="3"
                ry="3"
                width="500"
                x="0"
                y="345"
              />
            </clippath>
            <lineargradient
              id="v0huufs-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#2d2d2d"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="3s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#717171"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="3s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#2d2d2d"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="3s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="common__StBox-sc-16ufsj2-0 lcfJTu"
    >
      <svg
        aria-labelledby="v0huufs-aria"
        height="500"
        role="img"
        viewBox="0 0 500 500"
        width="500"
      >
        <title
          id="v0huufs-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#v0huufs-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#v0huufs-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="v0huufs-diff"
          >
            <rect
              height="100"
              rx="3"
              ry="3"
              width="500"
              x="0"
              y="0"
            />
            <rect
              height="100"
              rx="3"
              ry="3"
              width="500"
              x="0"
              y="115"
            />
            <rect
              height="100"
              rx="3"
              ry="3"
              width="500"
              x="0"
              y="230"
            />
            <rect
              height="100"
              rx="3"
              ry="3"
              width="500"
              x="0"
              y="345"
            />
          </clippath>
          <lineargradient
            id="v0huufs-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#2d2d2d"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="3s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#717171"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="3s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#2d2d2d"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="3s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Planets should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="common__StBox-sc-16ufsj2-0 lcfJTu"
      >
        <div
          class="common__StBoxItem-sc-16ufsj2-1 elAStK"
        >
          <div>
            Name: 
            Tatooine
          </div>
          <div>
            Climate: 
            Arid
          </div>
          <div>
            Orbital Period: 
            304
          </div>
          <div>
            Population: 
            120000
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="common__StBox-sc-16ufsj2-0 lcfJTu"
    >
      <div
        class="common__StBoxItem-sc-16ufsj2-1 elAStK"
      >
        <div>
          Name: 
          Tatooine
        </div>
        <div>
          Climate: 
          Arid
        </div>
        <div>
          Orbital Period: 
          304
        </div>
        <div>
          Population: 
          120000
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Planets should render wrapper 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="common__StBox-sc-16ufsj2-0 lcfJTu"
      >
        <div
          class="common__StBoxItem-sc-16ufsj2-1 elAStK"
        >
          <div>
            Name: 
            Tatooine
          </div>
          <div>
            Climate: 
            Arid
          </div>
          <div>
            Orbital Period: 
            304
          </div>
          <div>
            Population: 
            120000
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="common__StBox-sc-16ufsj2-0 lcfJTu"
      >
        <div
          class="common__StBoxItem-sc-16ufsj2-1 elAStK"
        >
          <div>
            Name: 
            Tatooine
          </div>
          <div>
            Climate: 
            Arid
          </div>
          <div>
            Orbital Period: 
            304
          </div>
          <div>
            Population: 
            120000
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="common__StBox-sc-16ufsj2-0 lcfJTu"
    >
      <div
        class="common__StBoxItem-sc-16ufsj2-1 elAStK"
      >
        <div>
          Name: 
          Tatooine
        </div>
        <div>
          Climate: 
          Arid
        </div>
        <div>
          Orbital Period: 
          304
        </div>
        <div>
          Population: 
          120000
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
